server: TestReportUsage failed under stress

SHA: https://github.com/cockroachdb/cockroach/commits/5a373445c0674f060a4bfe369ad290a0cacccb6c

Parameters:
```
TAGS=
GOFLAGS=-race
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
make stressrace TESTS=TestReportUsage PKG=github.com/cockroachdb/cockroach/pkg/server TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=982983&tab=buildLog

```
I181024 06:10:06.304822 55918 sql/event_log.go:126  [n1,client=127.0.0.1:46434,user=root] Event: "set_zone_config", target: 1, info: {Target:system Config:experimental_lease_preferences: [[+zone=somestring,+somestring], [+somestring]] Options: User:root}
I181024 06:10:06.366742 55918 sql/event_log.go:126  [n1,client=127.0.0.1:46434,user=root] Event: "create_table", target: 53, info: {TableName:somestring.public.somestring Statement:CREATE TABLE somestring.public.somestring (somestring INT, CONSTRAINT somestring CHECK (somestring > 1)) User:root}
I181024 06:10:06.368845 56024 storage/replica_command.go:300  [n1,split,s1,r20/1:/{Table/23-Max}] initiating a split of this range at key /Table/53 [r21]
I181024 06:10:06.430210 55501 storage/allocator_scorer.go:597  [n1,replicate,s1,r21/1:/{Table/53-Max}] nodeHasReplica(n1, [(n1,s1):1])=true
I181024 06:10:06.433632 55484 storage/allocator_scorer.go:597  [n1,replicate,s1,r21/1:/{Table/53-Max}] nodeHasReplica(n1, [(n1,s1):1])=true
I181024 06:10:06.557376 55918 sql/event_log.go:126  [n1,client=127.0.0.1:46434,user=root] Event: "set_cluster_setting", target: 0, info: {SettingName:cluster.organization Value:'Cockroach Labs - Production Testing' User:root}
I181024 06:10:06.676387 55918 sql/event_log.go:126  [n1,client=127.0.0.1:46434,user=root] Event: "set_cluster_setting", target: 0, info: {SettingName:cluster.organization Value:'Cockroach Labs - Production Testing' User:root}
I181024 06:10:06.810893 55918 sql/event_log.go:126  [n1,client=127.0.0.1:46434,user=root] Event: "set_cluster_setting", target: 0, info: {SettingName:cluster.organization Value:'Cockroach Labs - Production Testing' User:root}
I181024 06:10:06.969674 55918 sql/event_log.go:126  [n1,client=127.0.0.1:46434,user=root] Event: "set_cluster_setting", target: 0, info: {SettingName:cluster.organization Value:'Cockroach Labs - Production Testing' User:root}
I181024 06:10:07.104882 55918 sql/event_log.go:126  [n1,client=127.0.0.1:46434,user=root] Event: "set_cluster_setting", target: 0, info: {SettingName:cluster.organization Value:'Cockroach Labs - Production Testing' User:root}
I181024 06:10:07.233135 55918 sql/event_log.go:126  [n1,client=127.0.0.1:46434,user=root] Event: "set_cluster_setting", target: 0, info: {SettingName:cluster.organization Value:'Cockroach Labs - Production Testing' User:root}
I181024 06:10:07.382032 55918 sql/event_log.go:126  [n1,client=127.0.0.1:46434,user=root] Event: "set_cluster_setting", target: 0, info: {SettingName:cluster.organization Value:'Cockroach Labs - Production Testing' User:root}
I181024 06:10:07.529298 55918 sql/event_log.go:126  [n1,client=127.0.0.1:46434,user=root] Event: "set_cluster_setting", target: 0, info: {SettingName:cluster.organization Value:'Cockroach Labs - Production Testing' User:root}
I181024 06:10:07.650934 55918 sql/event_log.go:126  [n1,client=127.0.0.1:46434,user=root] Event: "set_cluster_setting", target: 0, info: {SettingName:cluster.organization Value:'Cockroach Labs - Production Testing' User:root}
I181024 06:10:07.777372 55918 sql/event_log.go:126  [n1,client=127.0.0.1:46434,user=root] Event: "set_cluster_setting", target: 0, info: {SettingName:cluster.organization Value:'Cockroach Labs - Production Testing' User:root}
I181024 06:10:08.287318 55207 util/stop/stopper.go:537  quiescing; tasks left:
2      [async] closedts-rangefeed-subscriber
1      [async] transport racer
I181024 06:10:08.287365 55464 kv/transport_race.go:113  transport race promotion: ran 26 iterations on up to 1490 requests
--- FAIL: TestReportUsage (4.46s)
	updates_test.go:547: expected 16 feature usage counts, got 17: map[errorcodes.0A000:10 errorcodes.22000:10 errorcodes.22012:10 unimplemented.#9148:10 unimplemented.pg_catalog.pg_stat_wal_receiver:10 othererror.builtins.go:10 othererror.22000.crdb_internal.set_vmodule():10 test.a:1 test.b:2 internalerror.:10 errorcodes.42601:10 errorcodes.blah:10 othererror.unknown:1 errorcodes.XX000:10 test.c:3 unimplemented.syntax.alter table rename constraint:10 unimplemented.syntax.interval with precision:10]

```

SHA: https://github.com/cockroachdb/cockroach/commits/704e5b8e9619a5015fa4ccbd13de2e5353221ca5

Parameters:
```
TAGS=
GOFLAGS=-race
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
make stressrace TESTS=TestReportUsage PKG=github.com/cockroachdb/cockroach/pkg/server TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1019946&tab=buildLog

```
I181120 07:17:04.197679 62668 storage/allocator_scorer.go:597  [n1,replicate,s1,r21/1:/{Table/53-Max}] nodeHasReplica(n1, [(n1,s1):1])=true
I181120 07:17:04.200044 62668 storage/allocator_scorer.go:597  [n1,replicate,s1,r21/1:/{Table/53-Max}] nodeHasReplica(n1, [(n1,s1):1])=true
I181120 07:17:04.437966 63254 sql/event_log.go:126  [n1,client=127.0.0.1:49814,user=root] Event: "set_cluster_setting", target: 0, info: {SettingName:cluster.organization Value:'Cockroach Labs - Production Testing' User:root}
I181120 07:17:04.525343 62853 storage/allocator_scorer.go:597  [n1,replicate,s1,r5/1:/System/ts{d-e}] nodeHasReplica(n1, [(n1,s1):1])=true
I181120 07:17:04.638508 63254 sql/event_log.go:126  [n1,client=127.0.0.1:49814,user=root] Event: "set_cluster_setting", target: 0, info: {SettingName:cluster.organization Value:'Cockroach Labs - Production Testing' User:root}
I181120 07:17:04.641945 63302 storage/split_queue.go:213  [n1,split,s1,r7/1:/Table/{SystemCon…-11}] initiating a split based on load at key /Table/3/1/1/2/1
E181120 07:17:04.642492 63302 storage/queue.go:837  [n1,split,s1,r7/1:/Table/{SystemCon…-11}] unable to split [n1,s1,r7/1:/Table/{SystemCon…-11}] at key /Table/3/1/1/2/1: cannot split range at key /Table/3/1/1/2/1
I181120 07:17:04.816980 63254 sql/event_log.go:126  [n1,client=127.0.0.1:49814,user=root] Event: "set_cluster_setting", target: 0, info: {SettingName:cluster.organization Value:'Cockroach Labs - Production Testing' User:root}
I181120 07:17:05.034835 63254 sql/event_log.go:126  [n1,client=127.0.0.1:49814,user=root] Event: "set_cluster_setting", target: 0, info: {SettingName:cluster.organization Value:'Cockroach Labs - Production Testing' User:root}
I181120 07:17:05.190000 63254 sql/event_log.go:126  [n1,client=127.0.0.1:49814,user=root] Event: "set_cluster_setting", target: 0, info: {SettingName:cluster.organization Value:'Cockroach Labs - Production Testing' User:root}
I181120 07:17:05.372461 63254 sql/event_log.go:126  [n1,client=127.0.0.1:49814,user=root] Event: "set_cluster_setting", target: 0, info: {SettingName:cluster.organization Value:'Cockroach Labs - Production Testing' User:root}
I181120 07:17:05.563969 63254 sql/event_log.go:126  [n1,client=127.0.0.1:49814,user=root] Event: "set_cluster_setting", target: 0, info: {SettingName:cluster.organization Value:'Cockroach Labs - Production Testing' User:root}
I181120 07:17:05.732195 63254 sql/event_log.go:126  [n1,client=127.0.0.1:49814,user=root] Event: "set_cluster_setting", target: 0, info: {SettingName:cluster.organization Value:'Cockroach Labs - Production Testing' User:root}
I181120 07:17:05.908558 63254 sql/event_log.go:126  [n1,client=127.0.0.1:49814,user=root] Event: "set_cluster_setting", target: 0, info: {SettingName:cluster.organization Value:'Cockroach Labs - Production Testing' User:root}
I181120 07:17:06.083487 63254 sql/event_log.go:126  [n1,client=127.0.0.1:49814,user=root] Event: "set_cluster_setting", target: 0, info: {SettingName:cluster.organization Value:'Cockroach Labs - Production Testing' User:root}
I181120 07:17:06.632974 62488 util/stop/stopper.go:537  quiescing; tasks left:
2      [async] closedts-rangefeed-subscriber
1      [async] transport racer
I181120 07:17:06.633202 62526 kv/transport_race.go:113  transport race promotion: ran 30 iterations on up to 1496 requests
I181120 07:17:06.636820 62488 util/stop/stopper.go:537  quiescing; tasks left:
2      [async] closedts-rangefeed-subscriber
I181120 07:17:06.637600 62488 util/stop/stopper.go:537  quiescing; tasks left:
1      [async] closedts-rangefeed-subscriber
--- FAIL: TestReportUsage (5.41s)
	updates_test.go:547: expected 16 feature usage counts, got 17: map[errorcodes.22012:10 errorcodes.42601:10 othererror.unknown:1 unimplemented.syntax.alter table rename constraint:10 errorcodes.XX000:10 errorcodes.blah:10 unimplemented.#9148:10 unimplemented.pg_catalog.pg_stat_wal_receiver:10 internalerror.:10 errorcodes.0A000:10 test.a:1 test.b:2 test.c:3 othererror.builtins.go:10 errorcodes.22000:10 othererror.22000.crdb_internal.set_vmodule():10 unimplemented.syntax.interval with precision:10]

```

--- a/pkg/server/updates_test.go
+++ b/pkg/server/updates_test.go
@@ -228,6 +228,9 @@ func TestReportUsage(t *testing.T) {
 	defer s.Stopper().Stop(context.TODO()) // stopper will wait for the update/report loop to finish too.
 	ts := s.(*TestServer)
 
+	// make sure the test's generated activity is the only activity we measure.
+	telemetry.GetAndResetFeatureCounts(false)
+
 	if _, err := db.Exec(fmt.Sprintf(`CREATE DATABASE %s`, elemName)); err != nil {
 		t.Fatal(err)
 	}
@@ -546,6 +549,7 @@ func TestReportUsage(t *testing.T) {
 		"errorcodes." + pgerror.CodeFeatureNotSupportedError: 10,
 		"errorcodes." + pgerror.CodeDivisionByZeroError:      10,
 	}
+
 	if expected, actual := len(expectedFeatureUsage), len(r.last.FeatureUsage); expected != actual {
 		t.Fatalf("expected %d feature usage counts, got %d: %v", expected, actual, r.last.FeatureUsage)
 	}

